src/hosts.cpp

```diff
@@ -85,6 +88,40 @@ code hosts::fetch(address& out) const
     ///////////////////////////////////////////////////////////////////////////
 }
 
+code hosts::fetch(address::list& out) const
+{
+    if (disabled_)
+        return error::not_found;
+
+    // Critical Section
+    ///////////////////////////////////////////////////////////////////////////
+    shared_lock lock(mutex_);
+
+    if (stopped_)
+        return error::service_stopped;
+
+    if (buffer_.empty())
+        return error::not_found;
+
+    auto out_count = std::min(buffer_.size(), pool_capacity_) /
+        static_cast<size_t>(pseudo_random(1, 20));
+
+    if (out_count == 0)
+        return error::success;
+
+    out.reserve(out_count);
+
+    size_t index = 0;
+    while (out_count--)
+        out.push_back(buffer_[index++]);
+
+    static auto generator = std::default_random_engine{
```

Comment:
Not sure we want a new engine here, but I need to look at the existing random utility comments.